ivy/functional/frontends/numpy/random/functions.py

```diff
@@ -61,3 +61,20 @@ def multinomial(n, pvals, size=None):
     else:
         num_samples = len(pvals)
     return ivy.multinomial(n, num_samples, batch_size=batch_size, probs=pvals)
+
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def permutation(x):
+    if isinstance(x, int):
+        new_array = ivy.arange(x)
+        ivy.random.shuffle(new_array)
+        return new_array
+    elif isinstance(x, str):
+        raise ValueError("x must be an integer or at least 1-dimensional")
+    elif x.ndim > 0:
+        new_array = x
+        ivy.random.shuffle(new_array)
+        return new_array
+    else:
+        raise ValueError("x must be an integer or at least 1-dimensional")
```

Comment:
a couple of things:
1.- There is no real need to do all the checks here, the backends will raise these errors.
2.- `ivy.shuffle` doesn't operate in-place, so you are not actually returning the shuffled array :slightly_smiling_face: 

probably this can all be summed up in 
```python
def permutation(x):
    if isinstance(x, int):
        x = ivy.arange(x)
    return ivy.shuffle(x)
```